### PR TITLE
improve documents of ACS

### PR DESCRIPTION
**Pull request by rjtsdl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by rjtsdl** [comment]
#4117 #4273

File: src/command_modules/azure-cli-acs/HISTORY.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ unreleased
 * correct preview regions
 * format default dns_name_prefix properly
 * optimize acs command output
+* improve documentions
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/HISTORY.rst]: History entry not required/desired for docs improvements. Only customer facing behavioral changes.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 2]
```diff
@@ -90,18 +90,18 @@ def _get_feature_in_preview_message():
...
-register_cli_argument('acs create', 'master_first_consecutive_static_ip', type=str, help=_get_feature_in_preview_message() + 'The first consecutive ip used to specify static ip block')
-register_cli_argument('acs create', 'master_storage_profile', help=_get_feature_in_preview_message(), **enum_choice_list(storageProfileTypes))
+register_cli_argument('acs create', 'master_osdisk_size', type=int, help=_get_feature_in_preview_message() + 'The disk size for master pool vms. Unit in GB. Default: corresponding vmsize disk size')
+register_cli_argument('acs create', 'master_vnet_subnet_id', type=str, help=_get_feature_in_preview_message() + 'The custom vnet subnet id. Note agent need to used the same vnet if master set. Default: ""')
+register_cli_argument('acs create', 'master_first_consecutive_static_ip', type=str, help=_get_feature_in_preview_message() + 'The first consecutive ip used to specify static ip block. Default: 10.240.255.5')
+register_cli_argument('acs create', 'master_storage_profile', help=_get_feature_in_preview_message() + 'Default: varies based on Orchestrator', **enum_choice_list(storageProfileTypes))
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: The CLI will automatically display default values if they are specified using the `default` kwarg or if they are specified as the default value in the command signature. Some of these make sense ("Default: latest version for location") but if it is a static value I'm not sure why you are putting them in the help text.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 3]
```diff
@@ -90,18 +90,18 @@ def _get_feature_in_preview_message():
...
+register_cli_argument('acs create', 'agent_osdisk_size', type=int, help=_get_feature_in_preview_message() + 'Set the default disk size for agent pools vms. Unit in GB. Default: corresponding vmsize disk size')
+register_cli_argument('acs create', 'agent_vnet_subnet_id', type=str, help=_get_feature_in_preview_message() + 'Set the default custom vnet subnet id for agent pools. Note agent need to used the same vnet if master set. Default: ""')
+register_cli_argument('acs create', 'agent_ports', type=validate_list_of_integers, help=_get_feature_in_preview_message() + 'Set the default ports exposed on the agent pools. Only usable for non-Kubernetes. Default: 8080,4000,80')
+register_cli_argument('acs create', 'agent_storage_profile', help=_get_feature_in_preview_message() + 'Set default storage profile for agent pools. Default: varies based on Orchestrator', **enum_choice_list(storageProfileTypes))
 
-register_cli_argument('acs create', 'windows', action='store_true', help='If true, deploy a windows container cluster.')
+register_cli_argument('acs create', 'windows', action='store_true', help='If true, set the default osType of agent pools to be windows.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: nit: Windows should be capitalized in the help text.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 4]
```diff
@@ -450,7 +450,7 @@ def acs_create(resource_group_name, deployment_name, name, ssh_key_value, dns_na
     :type master_storage_profile: str
     :param agent_profiles: AgentPoolProfiles used to describe agent pools
     :type agent_profiles: dict
-    :param agent_count: The number of agents for the cluster.  Note, for
+    :param agent_count: Set default number of agents for the agent pools.  Note, for
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: The note should arguably go in the `help.py` file as a `long-summary` entry.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py (hunk 2 above)
Comment (rjtsdl) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: I should have more static values to the function prototype. Will fix.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 4 above)
Comment (rjtsdl) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: I moved the long version to _params.py file. I am very clear the strategy's here to use `long-summary`, after i get familiarized, i probably need to do that across the code.

**Pull request merged**